### DISCUSSION TITLE
Follow Trusted Locations

### DISCUSSION THREAD

**Discussion by MicahBCode** [curl/curl#13130] [Q&A] [answered]
Hello everyone,

Yesterday we have updated our server from CentOS 7 to Alma Linux 8 which resulted in an update of libcurl from version 7.29.0 to 7.61.2. We noticed, that Basic Auth credentials, that were provided using the -u option are not being sent to the followed Location headers in case of a 30x response. Therefore we had to update to use the --location-trusted option in addition.

We noticed this problem in a case were the request was sent to http://example.com/scripts/myscript.php. This resulted in a 301 response with Location header set to https://example.com/scripts/myscript.php. In this case the Credentials then were not sent along, even though the host was exactly the same but only the protocol has changed from http to https. In such a case, shouldn't the Authorization header also be sent in the retry where it tries to reach the Location from the first response?

**Comment by bagder** [comment, marked as answer]
> shouldn't the Authorization header also be sent in the retry where it tries to reach the Location from the first response?

No. Because the "origin" is the protocol, hostname and port number triplet. An a shared server, the HTTP service could be run by a completely different person and organization than who runs the HTTPS service.

**Reply by MicahBCode** [reply]
Ahh, sure! Thanks for the fast reply :)
